modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -71,13 +70,23 @@ defineADValidParams(
         "the old creep strain will be reset to the function value at the beginning of the "
         "timestep. Used for testing purposes only.");
 
-    params.addParam<bool>("verbose", false, "Flag to add verbose output"););
+    params.addParam<bool>("verbose", false, "Flag to add verbose output");
+
+    params.addRangeCheckedParam<unsigned int>(
```

Comment:
I'm puzzled by the `stress_index` parameter.  This seems to be something that should be hard-coded in the derived class, rather than user settable.  That's how it was done in the UserObject version of this, at least.

Reply:
Since the derivative of the residual with the stress is required, I have the `stress_index` to key which ROM input is the stress. There isn't a case where this isn't always the same, but I didn't want a number floating around the conditional checks, and it is more clear with a name. If it does end up changing, it can't really be hard coded since it'll change in the constructor, unless I have a separate virtual method. It seemed easier to have an input parameter with a default that could be changed in the inherited class, much like the initial conditions.

Reply:
I guess I could just hard code it in the base class, instead of having a user settable parameter...that does make more sense